src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.cs

```diff
@@ -25,38 +26,109 @@ public partial class CborWriter : IDisposable
         private int _frameOffset = 0; // buffer offset particular to the current data item context
         private bool _isTagContext = false; // true if writer is expecting a tagged value
 
-        // Map-specific conformance book-keeping
-        private int? _currentKeyOffset = null;
-        private int? _currentValueOffset = null;
-        // State required for sorting map keys
-        private bool _keysRequireSorting = false;
-        private List<KeyValueEncodingRange>? _keyValueEncodingRanges = null;
-        // State required for checking key uniqueness
-        private HashSet<(int Offset, int Length)>? _keyEncodingRanges = null;
+        // Map-specific book-keeping
+        private int? _currentKeyOffset = null; // offset for the current key encoding
+        private int? _currentValueOffset = null; // offset for the current value encoding
+        private bool _keysRequireSorting = false; // tracks whether key/value pair encodings need to be sorted
+        private List<KeyValuePairEncodingRange>? _keyValuePairEncodingRanges = null; // all key/value pair encoding ranges
+        private HashSet<(int Offset, int Length)>? _keyEncodingRanges = null; // all key encoding ranges up to encoding equality
 
-        public CborWriter(CborConformanceLevel conformanceLevel = CborConformanceLevel.Lax, bool encodeIndefiniteLengths = false, bool allowMultipleRootLevelValues = false)
+        /// <summary>
+        ///   The <see cref="CborConformanceLevel"/> used by this writer.
+        /// </summary>
+        public CborConformanceLevel ConformanceLevel { get; }
+
+        /// <summary>
+        ///   True if the writer automatically converts indefinite-length encodings into definite-length equivalents.
+        /// </summary>
+        public bool ConvertIndefiniteLengthEncodings { get; }
```

Comment:
Boolean properties are a bit different, per https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Property and https://github.com/dotnet/dotnet-api-docs/wiki/Property-Value.

```suggestion
        /// <summary>
        ///   Gets a value that indicates whether the writer automatically converts indefinite-length encodings into definite-length equivalents.
        /// </summary>
        /// <value>
        ///   <see langword="true"/> if the writer automatically converts indefinite-length encodings into definite-length equivalents; otherwise, <see langword="false"/>.
        /// </value>
        public bool ConvertIndefiniteLengthEncodings { get; }
```